### ISSUE TITLE
Account name or password is not correct.

### ISSUE THREAD

**Issue by Source61** [slawkens/myaac#147]
### Server configuration
- Operating System: Debian 10
- Web Server (+ version): Apache/2.4.38
- PHP Version: 7.3.19-1~deb10u1
- Server name and version (for example: TFS 0.3): Nekiro's TFS 1.3 downgrade (7.72)
- MyAAC Version: 0.8.3

### Client configuration (Your Computer)
- Browser: Firefox
- Operating System: Windows 10

### Description:
Installation basically went fine except I had to modify 1 sql query to get it working with TFS 1.3 schema (changed column `id` to `number` in myaac query due to sql error - nekiro schema uses field `number` for id).
I could make new accounts, login to account manager, logout, and login again.

I'm not sure what's happened since I haven't touched myaac ever since, but logging in with account manager stopped working even though I can still login to the server, so nothing's changed in mysql or TFS.

I also created a new account with the same pw and got the same salt in mysql.

Reinstalling the AAC gave me temporary admin access again without having to login, but I tried logging in again in a private tab and got the same account name or password is not correct. Tried it about 10+ times, made sure caps was off, and got locked out more than once.

I've also tried setting db info in myaac's config.php, no change.

No errors in apache log.

### Steps To Reproduce:
Use Nekiro's TFS 1.3 schema, get an sql error during myaac installation, change the field name in the line of the error, no more errors, everything working, randomly can't login to the AAC even tho you can still login to the server.

**Comment by WalistonBelles** [comment]
Have you checked if you are using Salt? Your server may be using salt to encrypt your password, but your website to access an account does not use salt. Make a test, go to the login.php file and check if you are using salt to access your account on the site.

> ### Server configuration
> * Operating System: Debian 10
> * Web Server (+ version): Apache/2.4.38
> * PHP Version: 7.3.19-1~deb10u1
> * Server name and version (for example: TFS 0.3): Nekiro's TFS 1.3 downgrade (7.72)
> * MyAAC Version: 0.8.3
> 
> ### Client configuration (Your Computer)
> * Browser: Firefox
> * Operating System: Windows 10
> 
> ### Description:
> Installation basically went fine except I had to modify 1 sql query to get it working with TFS 1.3 schema (changed column `id` to `number` in myaac query due to sql error - nekiro schema uses field `number` for id).
> I could make new accounts, login to account manager, logout, and login again.
> 
> I'm not sure what's happened since I haven't touched myaac ever since, but logging in with account manager stopped working even though I can still login to the server, so nothing's changed in mysql or TFS.
> 
> I also created a new account with the same pw and got the same salt in mysql.
> 
> Reinstalling the AAC gave me temporary admin access again without having to login, but I tried logging in again in a private tab and got the same account name or password is not correct. Tried it about 10+ times, made sure caps was off, and got locked out more than once.
> 
> I've also tried setting db info in myaac's config.php, no change.
> 
> No errors in apache log.
> 
> ### Steps To Reproduce:
> Use Nekiro's TFS 1.3 schema, get an sql error during myaac installation, change the field name in the line of the error, no more errors, everything working, randomly can't login to the AAC even tho you can still login to the server.

**Comment by Source61** [comment]
@WalistonBelles
Hey, appreciate the response.
Some kind of encryption is for sure being used, however myaac seem to detect and use this themselves, since
1) I've only registered accounts through myaac and all the passwords turns out encrypted, and
2) I tried registering a new account with the same password and got the same encrypted result

And altho I'm not new to development, I'm new to aac's and tfs (last aac I ever used was nicaw back in the 2000's) and have no idea where to look for these things. You mentioned go to acc login, but that doesn't do anything. I tried searching for 'salt' both in tfs and myaac config and got 0 results though.

I also tried logging in using the encrypted pw from mysql in case like you said myaac isn't using salting or something like that, but it didn't work.

Also you can try it yourself here: http://server.opentibia.dev/
You'll be able to register and use your new acc to login to the server with cip client or otclient, but you won't be able to login.

I suspect the issue must have come from my 1 sql edit since nobody else has this issue, but I'm not sure how I can make myaac work with Nekiro's TFS 1.3 if I can't make this sql edit.

Btw the sql edit I made is in system/libs/pot/OTS_Account.php line 140. It still does isset($id) since that's just the varname derived from the function args, but the text output is `number` instead of `id`.

**Comment by Source61** [comment]
My bad, sort of. I had to edit another

**Comment by slawkens** [comment]
Did you made any changes to support this engine (7.72 Downgrade)?

Maybe you could do a PR? Would be nice to see what you changed.

**Comment by Source61** [comment]
> 
> 
> Did you made any changes to support this engine (7.72 Downgrade)?
> 
> Maybe you could do a PR? Would be nice to see what you changed.

Unfortunately I didn't see your reply until now and switched the same or next day to Znote AAC.
I believe the minor changes I made was to two queries in system/libs/pot/OTS_Account.php.
I think I just replaced the column name (or id) with number, but I don't remember, I went down two different paths and I think one of them didn't work as intended, so I can't recall properly.
